src/main/java/picocli/CommandLine.java

```diff
@@ -5484,32 +5483,32 @@ public Help create(CommandSpec commandSpec, Help.ColorScheme colorScheme) {
                 Callable<?> callable = Callable.class.cast(cls.getConstructor(Object.class, Object.class).newInstance(null, null));
                 try { return (T) callable.call(); }
                 catch (Exception ex) { throw new InitializationException("Error in Groovy closure: " + ex); }
-
             }
-            if (cls.isInterface() && Collection.class.isAssignableFrom(cls)) {
-                if (List.class.isAssignableFrom(cls)) {
-                    return cls.cast(new ArrayList<Object>());
-                } else if (SortedSet.class.isAssignableFrom(cls)) {
-                    return cls.cast(new TreeSet<Object>());
-                } else if (Set.class.isAssignableFrom(cls)) {
-                    return cls.cast(new LinkedHashSet<Object>());
-                } else if (Queue.class.isAssignableFrom(cls)) {
-                    return cls.cast(new LinkedList<Object>()); // ArrayDeque is only available since 1.6
+            if (cls.isInterface()) {
+                if (Collection.class.isAssignableFrom(cls)) {
+                    if (List.class.isAssignableFrom(cls)) {
+                        return cls.cast(new ArrayList<Object>());
+                    } else if (SortedSet.class.isAssignableFrom(cls)) {
+                        return cls.cast(new TreeSet<Object>());
+                    } else if (Set.class.isAssignableFrom(cls)) {
+                        return cls.cast(new LinkedHashSet<Object>());
+                    } else if (Queue.class.isAssignableFrom(cls)) {
+                        return cls.cast(new LinkedList<Object>()); // ArrayDeque is only available since 1.6
+                    } else {
+                        return cls.cast(new ArrayList<Object>());
+                    }
+                }
+                if (SortedMap.class.isAssignableFrom(cls)) {
+                    return cls.cast(new TreeMap<Object, Object>());
+                }
+                if (Map.class.isAssignableFrom(cls)) {
+                    return cls.cast(new LinkedHashMap<Object, Object>());
                 }
-                return cls.cast(new ArrayList<Object>());
-            }
-            if (Map.class.isAssignableFrom(cls)) {
-                try { // if it is an implementation class, instantiate it
-                    return cls.cast(cls.getDeclaredConstructor().newInstance());
-                } catch (Exception ignored) { }
-                return cls.cast(new LinkedHashMap<Object, Object>());
             }
+            Constructor<T> constructor = cls.getDeclaredConstructor();
             try {
-                @SuppressWarnings("deprecation") // Class.newInstance is deprecated in Java 9
-                T result = cls.newInstance();
-                return result;
-            } catch (Exception ex) {
-                Constructor<T> constructor = cls.getDeclaredConstructor();
+                return constructor.newInstance();
+            } catch (IllegalAccessException ex) {
```

Comment:
...but I do like this idea of setting accessible=true only in the catch block of IllegalAccessException, but this try/catch should remain in the catch(Exception) block of `Class.newInstance`.

Reply:
Is this what you want?

```java
            try {
                @SuppressWarnings("deprecation") // Class.newInstance is deprecated in Java 9
                T result = cls.newInstance();
                return result;
            } catch (IllegalAccessException ex) {
                Constructor<T> constructor = cls.getDeclaredConstructor();
                constructor.setAccessible(true);
                return constructor.newInstance();
            }
```

Reply:
Almost, I had this in mind:

```java
try {
    @SuppressWarnings("deprecation") // Class.newInstance is deprecated in Java 9
    T result = cls.newInstance();
    return result;
} catch (Exception ex) {
    // TODO log the error at debug level
    Constructor<T> constructor = cls.getDeclaredConstructor();
    try {
        return constructor.newInstance();
    } catch (IllegalAccessException ex) {
        constructor.setAccessible(true);
        return constructor.newInstance();
    }
}
```
but not sure how much of an improvement this is.
The logging at DEBUG level when `Class.newInstance` fails feels like a nice improvement.

Reply:
I can see how debugging can help learn the exact differences between the 2 methods. I will modify the PR later today & resubmit.

Reply:
@rgoldberg Did you have a chance to look at this?

Reply:
Sorry. Couldn't look at this for a while. Will implement the changes soon.

Reply:
Pushed requested changes